cpp/src/fil/infer.cu

```diff
@@ -223,44 +205,54 @@ struct tree_aggregator_t<NITEMS, TREE_PER_CLASS_FEW_CLASSES> {
     acc += single_tree_prediction;
   }
 
+  // block-reduce the best candidate class and write it out to global memory
+  static __device__ __forceinline__ void block_reduce_write_out(
```

Comment:
This function is used in multiple classes. So it's better to either make it a non-static member of the base class (and reduce the number of parameters), or make it a standalone function that is not a member of any specific class.

Reply:
Let's discuss it tomorrow during the 1:1, as it has implications for softmax